Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -1143,4 +1175,53 @@ def problem(text)
       @problems << text
     end
   end
+
+  class TapAuditor
+    attr_reader :name, :path, :tap_audit_exceptions, :problems
+
+    def initialize(tap, strict:)
+      @name                 = tap.name
+      @path                 = tap.path
+      @tap_audit_exceptions = tap.audit_exceptions
+      @problems             = []
+    end
+
+    def audit
+      audit_json_files
+      audit_tap_audit_exceptions
+    end
+
+    def audit_json_files
+      Pathname.glob(@path/"**/*.json").each do |file|
+        JSON.parse file.read
+      rescue JSON::ParserError
+        problem "#{file.to_s.delete_prefix("#{@path}/")} contains invalid JSON"
+      end
+    end
+
+    def audit_tap_audit_exceptions
+      @tap_audit_exceptions.each do |list_name, formula_names|
+        formula_names = formula_names.keys if formula_names.is_a? Hash
+
+        invalid_formulae = []
+        formula_names.each do |name|
```

Comment:
Might be worth handling the case where this is neither a `Hash` or an `Array`?

Reply:
Not opposed, but not sure what the benefit is. It should always be one of these if it's a valid list. If it's not, it will return `nil`.

Reply:
It should be but: it might not be so adding more error handling in that case is probably no bad thing 😁

Reply:
So how would you recommend handling it? Return `false` with a warning?

Reply:
Oh, wait, I was confused about which section this was. I agree, this should be handled. `audit` will now complain if an exception file is not a JSON array or JSON object